packages/roosterjs-content-model/lib/modelApi/selection/deleteSelections.ts

```diff
@@ -256,6 +266,33 @@ const DeleteSelectionSteps: DeleteSelectionStep[] = [
     deleteSelectionStep3,
 ];
 
+function saveDeletedElements(
+    block: ContentModelBlock | undefined,
+    context: DeleteSelectionContext,
+    tableContext: TableSelectionContext | undefined,
+    deleteResult: ContentModelDocument
+) {
+    if (block) {
+        const newBlock = cloneBlock(block);
```

Comment:
I feel using cloneBlock is too much overhead. It will clone the whole block including all segments even they are not selected.

We can just move the deleted content into target model. For blocks, just move and push into target model as a block. For segment, may need to create a new Paragraph, with the same format of source paragraph.